mail_edit/static/src/js/mail_edit.js

```diff
@@ -0,0 +1,75 @@
+/* © 2014-2017 Sunflower IT <www.sunflowerweb.nl>*/
+
+ odoo.define('mail_edit.mail_edit', function (require) {
+"use strict";
+
+    var Message = require('mail.model.Message');
+    var ThreadWidget = require('mail.widget.Thread');
+    var Dialog = require('web.Dialog');
+
+    ThreadWidget.include({
+        events: _.defaults({
+            "click .o_mail_edit": "_onClickMessageEdit",
+            "click .o_mail_delete": "_onClickMessageDelete",
+        }, ThreadWidget.prototype.events),
+
+        _onClickMessageEdit: function(event) {
+            var self = this;
+            var do_action = self.do_action,
+                msg_id = $(event.currentTarget).data('message-id');
+
+            self._rpc({
+                route: "/web/action/load",
+                params: {
+                    action_id: "mail_edit.mail_edit_action",
+                },
+            })
+            .done(function (action) {
+                action.res_id = msg_id;
+                self.do_action(action, {
+                    on_close: function () {
+                        location.reload();
+                        //self.trigger_up('refresh_on_fly');
+                        //this.reload.bind(this)
+                        //self.trigger_up('reload', { keepChanges: true });
+                         //self.trigger.bind(self, 'need_refresh')
```

Comment:
I'll take a look at this.

Reply:
@Kiplangatdan That was more work than I thought, Odoo has many routines for live adding of new messages, but not for updating existing messages alas. Can you handle the remaining points, then I think we're good to go.

Reply:
@thomaspaulb Reload is now even better than before! 
We can now push it for review.